core/modules/ckeditor5/js/ckeditor5.js

```diff
@@ -301,10 +301,47 @@
       beforeElement.innerHTML = beforeAttachValue;
       beforeAttachValue = Backdrop.ckeditor5.elementGetHtml(beforeElement.content);
 
-      // Then run both strings through the same whitespace formatting.
-      const formattedBeforeValue = Backdrop.ckeditor5.formatHtml(beforeAttachValue);
-      const formattedAfterValue = Backdrop.ckeditor5.formatHtml(afterAttachValue);
-      return formattedBeforeValue !== formattedAfterValue;
+      // Then run both strings through the same whitespace formatting, using
+      // formatHtml(). Wrap both strings with a temporary <temp> tag, to allow
+      // childNodes (which is used later when comparing the two strings) to work
+      // on them.
+      const formattedBeforeValue = document.createElement('temp');
+      formattedBeforeValue.innerHTML = Backdrop.ckeditor5.formatHtml(beforeAttachValue);
+      const formattedAfterValue = document.createElement('temp');
```

Comment:
Just as precaution, lets use a defined tag (`<div>`) in case it causes any kind of weird browser behaviors.

```suggestion
      // formatHtml(). Wrap both strings with a temporary <div> tag, to allow
      // childNodes (which is used later when comparing the two strings) to work
      // on them.
      const formattedBeforeValue = document.createElement('div');
      formattedBeforeValue.innerHTML = Backdrop.ckeditor5.formatHtml(beforeAttachValue);
      const formattedAfterValue = document.createElement('div');
```